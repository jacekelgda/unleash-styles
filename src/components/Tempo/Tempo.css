@media screen and (min-width: 768px) {
  .Tempo {
    max-width: 33%;
    width: 372px;
    margin-left: auto;
  }
}

.Tempo_title {
  font: 400 14px 'Montserrat', sans-serif;
  text-transform: uppercase;
  color: #b8b7bc;
  margin: 0 0 20px;
}
@media screen and (max-width: 767px) {
  .Tempo_title {
    margin-top: 40px;
  }
}
.Tempo_graph {
  width: 100%;
  overflow: hidden;
}
.Tempo_graph_img {
  min-width: 100%;
}

.Tempo_goals {
  list-style: none;
  margin: 0;
  padding: 0;
}
@media screen and (min-width: 1024px) {
  .Tempo_goals {
    display: -webkit-flex;
    display: -moz-flex;
    display: -ms-flex;
    display: -o-flex;
    display: flex;
    justify-content: space-between;
  }
}

@media screen and (max-width: 1023px) {
  /* Goals vertical aligned on Mobile & Tablets */
  .Tempo_goals {
    margin-top: 1em;
    padding: .75em 0 .5em;
    border-top: 1px solid #e9edee;
    border-bottom: 1px solid #e9edee;
  }
  .Tempo_goal {
    display: -webkit-flex;
    display: -moz-flex;
    display: -ms-flex;
    display: -o-flex;
    display: flex;
    justify-content: space-between;
  }
}

.Tempo_goal_label {
  display: block;
  font: 14px/3 'Montserrat', sans-serif;
  text-transform: uppercase;
  color: #b8b7bc;
}
.Tempo_goal_amount {
  display: block;
  font: 20px 'Montserrat', sans-serif;
  color: #242424;
}
