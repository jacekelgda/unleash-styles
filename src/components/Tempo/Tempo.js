/** @flow */
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styles from './Tempo.css'

/**
 * Tempo component to host Unleash Tempo graph
 */
class Tempo extends Component {
  static defaultProps: Object

  render () {
    const {title, children} = this.props
    return (
      <div className={styles.Tempo}>

        <h4 className={styles.Tempo_title}>{title}</h4>
        {children}
      </div>
    )
  }
}

Tempo.defaultProps = {
  paths: 0,
  stepsOpened: 0,
  stepsClosed: 0
}

Tempo.propTypes = {
  /** Component title */
  title: PropTypes.string.isRequired
}

export default Tempo
