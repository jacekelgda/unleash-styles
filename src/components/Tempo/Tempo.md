Tempo component:

    <Tempo title="Unleash Tempo">
      <Graph data={
        [{tempo: 1},
        {tempo: 2},
        {tempo: 3},
        {tempo: 2},
        {tempo: 3},
        {tempo: 4},
        {tempo: 4},
        {tempo: 5},
        {tempo: 1},
        {tempo: 1},
        {tempo: 1},
        {tempo: 1},
        {tempo: 1},
        {tempo: 1},
        {tempo: 1},
        {tempo: 1},
        {tempo: 1},
        {tempo: 1},
        {tempo: 1},
        {tempo: 6}]
        } />
      <TempoLegend>
        <TempoLegendItem label="Paths" value={20} />
        <TempoLegendItem label="Open Steps" value={130} />
        <TempoLegendItem label="Finished Steps" value={100} />
      </TempoLegend>
    </Tempo>
