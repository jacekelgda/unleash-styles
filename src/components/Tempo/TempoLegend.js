/** @flow */
import React from 'react'
import styles from './Tempo.css'

/**
 * TempoLegend component hosts TempoLegendItems
 */
const TempoLegend = (props: Object) => {
  return (
    <ul className={styles.Tempo_goals}>
      {props.children}
    </ul>
  )
}

export default TempoLegend
