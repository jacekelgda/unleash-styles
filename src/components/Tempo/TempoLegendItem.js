/** @flow */
import React from 'react'
import styles from './Tempo.css'
import PropTypes from 'prop-types'

/**
 * TempoLegendItems show different data related to Tempo
 */
const TempoLegendItem = (props: Object) => {
  return (
    <li className={styles.Tempo_goal}>
      <span className={styles.Tempo_goal_label}>{props.label}</span>
      <span className={styles.Tempo_goal_amount}>{props.value}</span>
    </li>
  )
}

TempoLegendItem.propTypes = {
  label: PropTypes.string,
  value: PropTypes.number
}

export default TempoLegendItem
