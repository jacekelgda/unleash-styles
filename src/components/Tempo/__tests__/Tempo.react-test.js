/* global test, expect */
import React from 'react'
import { shallow } from 'enzyme'
import Tempo from '../Tempo'
import TempoLegend from '../TempoLegend'
import TempoLegendItem from '../TempoLegendItem'

test('Tempo renders correctly', () => {
  const wrapper = shallow(<Tempo title='Title' />)
  expect(wrapper).toMatchSnapshot()
})

test('Tempo renders correctly with children', () => {
  const wrapper = shallow(
    <Tempo title='Title'>
      <div>Hello</div>
      <TempoLegend>
        <TempoLegendItem label='Test' value={3} />
      </TempoLegend>
    </Tempo>
  )
  expect(wrapper).toMatchSnapshot()
})
