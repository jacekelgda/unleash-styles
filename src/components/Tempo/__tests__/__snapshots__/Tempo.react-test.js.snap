// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tempo renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="Tempo"
>
    <h4
        className="Tempo_title"
    >
        Title
    </h4>
</div>,
  "nodes": Array [
    <div
      className="Tempo"
>
      <h4
            className="Tempo_title"
      >
            Title
      </h4>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Tempo
        paths={0}
        stepsClosed={0}
        stepsOpened={0}
        title="Title"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Tempo {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "paths": 0,
          "stepsClosed": 0,
          "stepsOpened": 0,
          "title": "Title",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="Tempo"
>
          <h4
                    className="Tempo_title"
          >
                    Title
          </h4>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="Tempo"
>
          <h4
                    className="Tempo_title"
          >
                    Title
          </h4>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Tempo
    paths={0}
    stepsClosed={0}
    stepsOpened={0}
    title="Title"
/>,
}
`;

exports[`Tempo renders correctly with children 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="Tempo"
>
    <h4
        className="Tempo_title"
    >
        Title
    </h4>
    <div>
        Hello
    </div>
    <TempoLegend>
        <TempoLegendItem
            label="Test"
            value={3}
        />
    </TempoLegend>
</div>,
  "nodes": Array [
    <div
      className="Tempo"
>
      <h4
            className="Tempo_title"
      >
            Title
      </h4>
      <div>
            Hello
      </div>
      <TempoLegend>
            <TempoLegendItem
                  label="Test"
                  value={3}
            />
      </TempoLegend>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Tempo
        paths={0}
        stepsClosed={0}
        stepsOpened={0}
        title="Title"
>
        <div>
                Hello
        </div>
        <TempoLegend>
                <TempoLegendItem
                        label="Test"
                        value={3}
                />
        </TempoLegend>
</Tempo>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Tempo {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": Array [
            <div>
              Hello
</div>,
            <TempoLegend>
              <TempoLegendItem
                            label="Test"
                            value={3}
              />
</TempoLegend>,
          ],
          "paths": 0,
          "stepsClosed": 0,
          "stepsOpened": 0,
          "title": "Title",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="Tempo"
>
          <h4
                    className="Tempo_title"
          >
                    Title
          </h4>
          <div>
                    Hello
          </div>
          <TempoLegend>
                    <TempoLegendItem
                              label="Test"
                              value={3}
                    />
          </TempoLegend>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="Tempo"
>
          <h4
                    className="Tempo_title"
          >
                    Title
          </h4>
          <div>
                    Hello
          </div>
          <TempoLegend>
                    <TempoLegendItem
                              label="Test"
                              value={3}
                    />
          </TempoLegend>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Tempo
    paths={0}
    stepsClosed={0}
    stepsOpened={0}
    title="Title"
>
    <div>
        Hello
    </div>
    <TempoLegend>
        <TempoLegendItem
            label="Test"
            value={3}
        />
    </TempoLegend>
</Tempo>,
}
`;
